test/unit/spec/common/cellComponents/tabs/jobStatusTable-Spec.js

```diff
@@ -570,77 +900,274 @@ define([
+                                        container
+                                            .querySelector(`[data-target="${input.retry_parent}"]`)
+                                            .click();
+                                        retryResponse({
+                                            bus: ctx.jobManager.bus,
+                                            retryParent,
+                                            retry: input,
+                                        });
+                                    } else {
+                                        updateState({
+                                            bus: ctx.jobManager.bus,
+                                            input,
+                                        });
+                                    }
+                                    // send the update for the batch parent
+                                    updateState({
+                                        bus: ctx.jobManager.bus,
+                                        input: updatedBatchJob,
+                                        jobId: updatedBatchJob.job_id,
+                                    });
+                                }
                             );
-                            this.input.meta.paramsRegex = test.regex;
 
-                            spyOn(this.jobManager, 'removeListener').and.callThrough();
-                            spyOn(this.jobManager, 'runHandler').and.callThrough();
-                            await TestUtil.waitForElementChange(this.row, () => {
-                                updateInfo(this, test.input);
+                            index++;
+                            const updatedRow = container.querySelector(
+                                '#job_' + input.retry_parent
+                            );
+                            _checkRowStructure(
+                                updatedRow,
+                                retryParent,
+                                update.expectedRow || input
+                            );
+                            expect(ctx.jobManager.updateModel).toHaveBeenCalled();
+                            expect(
+                                ctx.jobManager.model.getItem(`exec.jobs.byId.${input.job_id}`)
+                            ).toEqual(input);
+                            expect(container.querySelectorAll('.batchUpdate').length).toEqual(
+                                index
+                            );
+                            return new Promise.resolve(index);
+                        };
+
+                        const doUpdates = (startIndex, ctx) => {
+                            updateTableLoop(startIndex, ctx).then((ix) => {
+                                if (ix < updates.length) {
+                                    doUpdates(ix, ctx);
+                                } else {
+                                    console.log('finished my promises!');
```

Comment:
This test gets to here and then it just waits until it times out. I can't work out how to get it to finish properly.

Reply:
It looks like `doUpdates` doesn't return anything, so there's nothing for the Promise below to resolve on. Maybe just returning the `updateTableLoop` result would work?

Reply:
Or maybe restructure so you don't have to do it recursively? Instead of using the `doUpdates` function at all, it could be something like:

```
let idx = 0;
for (idx < updates.length) {
    idx = await updateTableLoop(idx, ctx);
    if (finishedIdx >= updates.length) {
        break;
    }
}
```
Or similar, and make the whole test case `async`?
Sorry if you've tried that already.